src/actions/searchActions.js

```diff
@@ -0,0 +1,57 @@
+import actionTypes from "./actionTypes";
+import {
+  API_KEY
+} from "../API_Key";
+
+export const setLoading = () => ({
+  type: actionTypes.SET_LOADING_STATE,
+});
+
+export const saveMoviesList = moviesList => ({
+  type: actionTypes.MOVIES_LIST_RECEIVED,
+  payload: moviesList
+});
+
+export const saveFullDescriptionMovie = fullDescription => ({
+  type: actionTypes.FULL_DESCRIPTION_MOVIE_RECEIVED,
+  payload: fullDescription
+});
+
+export const openDetailedPage = movie => ({
+  type: actionTypes.OPEN_DETAILED_PAGE,
+  payload: {
+    movie,
+  }
+});
+
+export const addToFavList = movie => ({
+  type: actionTypes.ADD_TO_FAV_MOVIES_LIST,
+  payload: movie,
+});
+
+export const deleteFromFavList = movie => ({
+  type: actionTypes.DELETE_FROM_FAV_MOVIES_LIST,
+  payload: movie,
+});
+
+export const fetchMovies = text => dispatch => {
+  dispatch(setLoading());
+
+  fetch(`https://www.omdbapi.com/?apikey=${API_KEY}&s=${text}&type=movie`)
+    .then(response => response.json())
+    .then(result => {
+      dispatch(saveMoviesList(result.Search));
+    })
+    .catch(err => console.log(err));
```

Comment:
Small improvement just to show you how you can do it
```suggestion
    .catch(console.log);
```